src/focus-camera/focus-first-residue.ts

```diff
@@ -60,12 +60,44 @@ export class FocusFirstResidue implements FocusFactoryI {
             imageRender.canvas3d.camera.setState(state);
         };
     }
-    getAxesToFlip(positions: Float32Array, origin: Vec3, up: Vec3, normalDir: Vec3) {
-        const toYAxis = calculateDisplacement(positions, origin, normalDir);
-        const toXAxis = calculateDisplacement(positions, origin, up);
+    getAxesToFlip(position: Vec3, origin: Vec3, up: Vec3, normalDir: Vec3) {
+        const toYAxis = calculateDisplacement(position, origin, normalDir);
+        const toXAxis = calculateDisplacement(position, origin, up);
         const Axes: string[] = [];
-        if (toYAxis[0] < 0) Axes.push('aroundY');
-        if (toXAxis[0] < 0) Axes.push('aroundX');
+        if (toYAxis < 0) Axes.push('aroundY');
+        if (toXAxis < 0) Axes.push('aroundX');
         return Axes;
     }
+}
+
+function getFirstResidueOrAveragePosition(structure: Structure, caPositions: Float32Array): Vec3 {
+    //TODO Is this the best way to test single chain?
+    // if only one chain => first residue coordinates
+    if (structure.units.length === 1) {
+        return Vec3.create(caPositions[0], caPositions[1], caPositions[2]);
+    } else {
+    // if more than one chain => average of coordinates of the first chain
+        const tmpMatrixPos = Vec3.zero();
+        const AtomIndexs = structure.units[0].elements;
+        const firstChainPositions = [];
+        for (let i = 0; i < AtomIndexs.length; i++) {
+            const coordinates = structure.units[0].conformation.position(AtomIndexs[i], tmpMatrixPos);
```

Comment:
Might be a good idea to cache the `structure.units[0].conformation.position` outside the loop.